src/comestible_inv_area.cpp

```diff
@@ -0,0 +1,275 @@
+                veh = &vp->vehicle();
+                veh_part = vp->part_index();
+            } else {
+                veh = nullptr;
+                veh_part = -1;
+            }
+            if( veh_part >= 0 ) {
+                desc[0] = veh->name;
+                is_valid_location = true;
+            } else {
+                veh = nullptr;
+                is_valid_location = false;
+                desc[0] = _( "No dragged vehicle!" );
+            }
+            break;
+        case ai::AIM_ALL:
+            desc[0] = _( "All 9 squares" );
+            is_valid_location = true;
+            break;
+        case ai::AIM_ALL_I_W:
+            desc[0] = _( "Around,Inv,Worn" );
+            is_valid_location = true;
+            break;
+        case ai::AIM_SOUTHWEST:
+        case ai::AIM_SOUTH:
+        case ai::AIM_SOUTHEAST:
+        case ai::AIM_WEST:
+        case ai::AIM_CENTER:
+        case ai::AIM_EAST:
+        case ai::AIM_NORTHWEST:
+        case ai::AIM_NORTH:
+        case ai::AIM_NORTHEAST:
+            if( const cata::optional<vpart_reference> vp = g->m.veh_at( pos ).part_with_feature( "CARGO",
+                    false ) ) {
+                veh = &vp->vehicle();
+                veh_part = vp->part_index();
+            } else {
+                veh = nullptr;
+                veh_part = -1;
+            }
+            is_valid_location = has_vehicle() || g->m.can_put_items_ter_furn( pos );
+            if( has_vehicle() ) {
+                desc[1] = vpart_position( *veh, veh_part ).get_label().value_or( "" );
+            }
+            // get graffiti or terrain name
+            desc[0] = g->m.has_graffiti_at( pos ) ?
+                      g->m.graffiti_at( pos ) : g->m.name( pos );
+        default:
+            break;
+    }
+
+    /* assemble a list of interesting traits of the target square */
+    // fields? with a special case for fire
+    bool danger_field = false;
+    const field &tmpfld = g->m.field_at( pos );
+    for( auto &fld : tmpfld ) {
+        const field_entry &cur = fld.second;
+        if( fld.first.obj().has_fire ) {
+            flags.append( _( " <color_white_red>FIRE</color>" ) );
+        } else {
+            if( cur.is_dangerous() ) {
+                danger_field = true;
+            }
+        }
+    }
+    if( danger_field ) {
+        flags.append( _( " DANGER" ) );
+    }
+
+    // trap?
+    const trap &tr = g->m.tr_at( pos );
+    if( tr.can_see( pos, g->u ) && !tr.is_benign() ) {
+        flags.append( _( " TRAP" ) );
+    }
+
+    // water?
+    static const std::array<ter_id, 8> ter_water = {
+        {t_water_dp, t_water_pool, t_swater_dp, t_water_sh, t_swater_sh, t_sewage, t_water_moving_dp, t_water_moving_sh }
+    };
+    auto ter_check = [pos]
+    ( const ter_id & id ) {
+        return g->m.ter( pos ) == id;
+    };
+    if( std::any_of( ter_water.begin(), ter_water.end(), ter_check ) ) {
+        flags.append( _( " WATER" ) );
+    }
+
+    // remove leading space
```

Comment:
Btw. we actually have a `trim` function for this (it removes spaces at the being and at the end).